### ISSUE TITLE
how to fix error /bin/sh: 2: xtensa-lx106-elf-gcc: not found

### ISSUE THREAD

**Issue by vandaipham** [espressif/ESP8266_MESH_DEMO#16]
Please, help me!
i have a error /bin/sh: 2: xtensa-lx106-elf-gcc: not found???

thanks very much!

**Comment by iam-black** [comment]
It seems that you are missing Xtensa crosstool-NG. See here https://github.com/esp8266/esp8266-wiki/wiki/Toolchain

Easiest way to compile without errors is to use official(?) VirtualBox_ESP8266 image where all tools and compilers are already present and set up. You can download it from here http://www.esp8266.com/viewtopic.php?f=6&t=490

**Comment by vandaipham** [comment]
thanks you very much!
tell me please is possible to transmit video server from a computer that is connected to one node in the network.??
firmware can be used for this purpose?

**Comment by iam-black** [comment]
Nope, this firmware in current state is not suitable for your needs. And I have doubts that ESP8266 is capable to do such tasks at all(depends on video bitrate of course).

**Comment by lossurdo** [comment]
I was trying to upload Lua firmware to **nodemcu 12e** and for that I would need to compile it. So I got the same error: `/bin/sh: 2: xtensa-lx106-elf-gcc: not found`.

My contribution to this thread is you to download a Docker Image with all tools you may need.

**Description:** 
ESP32, ESP8266, ESP8285 build environment of docker.

**URL:**
https://hub.docker.com/r/vowstar/esp8266/

**Chip support**
ESP8266
ESP8285
ESP31B
ESP32 series
ESP8268 and ESP8689 not tested yet.

**Tools**
xtensa-lx106-elf toolchain
xtensa-esp108-elf toolchain
xtensa-esp32-elf toolchain
esptool-ck
python and pyserial

After Docker Image installation I have used this command:

`docker run -ti --name esp-builder --device /dev/ttyUSB0:/dev/ttyUSB0 -v ~/Projects/2016:/build vowstar/esp8266 /bin/bash`

With this results:

> root@968e1ebacaeb:/build/nodemcu-firmware# make flash4m
> make -e FLASHOPTIONS="-fm dio -fs 32m -ff 40m" flashinternal
> make[1]: Entering directory '/build/nodemcu-firmware'
> make -C ./app flashinternal
> make[2]: Entering directory '/build/nodemcu-firmware/app'
> ../tools/esptool.py --port /dev/ttyUSB0 write_flash -fm dio -fs 32m -ff 40m 0x00000 ../bin/0x00000.bin 0x10000 ../bin/0x10000.bin
> esptool.py v1.2-dev
> Connecting...
> Running Cesanta flasher stub...
> Flash params set to 0x0240
> Writing 32768 @ 0x0... 32768 (100 %)
> Wrote 32768 bytes at 0x0 in 2.9 seconds (91.7 kbit/s)...
> Writing 385024 @ 0x10000... 385024 (100 %)
> Wrote 385024 bytes at 0x10000 in 33.5 seconds (92.0 kbit/s)...
> Leaving...
> make[2]: Leaving directory '/build/nodemcu-firmware/app'
> make[1]: Leaving directory '/build/nodemcu-firmware'

And it works!
